### PR TITLE
Javascript Capstone Group Project

### PR DESCRIPTION
**Pull request by Baayeh**
### In this project, we implemented the following tasks:

- Used JavaScript to make websites dynamic and build basic single-page apps.
- Used ES6 syntax.
- Used ES6 modules.
- Used callbacks and promises.
- Used webpack.
- Applied JavaScript best practices and language style guides in code.
- Used AAA pattern for unit tests.
- Wrote unit tests for a JavaScript app.
- Followed Gitflow.
- Solved simple git conflicts.
- Sent and receive data from an API.
- Used API documentation.
- Understand and use JSON.
- Made JavaScript code asynchronous.
- Performed code reviews for a team member.

### CODE REVIEW

File: src/modules/__test__/commentsCounter.test.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import commentsCounter from '../commentsCounter.js';
+
+describe('CommentsCounter', () => {
+  test('Array Length should be 4', () => {
+    const arr = [1, 2, 3, 4];
+    expect(commentsCounter(arr)).toBe(4);
+  });
+});
```
Comment (ichala) [inline src/modules/__test__/commentsCounter.test.js:8]: - [ ] Good Job testing all of the functions of the counters 🎉 but please note that you should test and cover all edge cases that can happen, I would suggest testing the case of having 0 items in the array where the expected result should be 0, by implementing that you will cover all of the edge cases 🙏 kindly fix that

File: src/modules/__test__/ItemsCounter.test.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import ItemsCounter from '../ItemsCounter.js';
```
Comment (ichala) [inline src/modules/__test__/ItemsCounter.test.js:1]: - [ ] Good Job testing all of the functions of the counters 🎉 but please note that you should test and cover all edge cases that can happen, I would suggest testing the case of having 0 items in the array where the expected result should be 0, by implementing that you will cover all of the edge cases 🙏 kindly fix that
- [ ] kindly try to use a correct descriptive message in your describe() method, in [ItemsCounter.test.js](https://github.com/Baayeh/second-capstone/pull/36/files#diff-c43a186005a06939bfe9495b50c3daa760cc2bc5d5c867440015c09b3a4f32ba) you described the test as `CommentsCounter` but it's not a comment counter , kindly fix it

File: src/index.html [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        Created by Kwasi & Daphine as Capstone Project for Module 2
+      </p>
+    </div>
+  </footer>
+
+
+  <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.0.2/dist/js/bootstrap.bundle.min.js"
+    integrity="sha384-MrcW6ZMFYlzcLA8Nl+NtUVF0sA7MsXsP1UyJoMp4YLEuNSfAP+JcXn/tWtIaxVXM"
+    crossorigin="anonymous"></script>
+  <script src="./index.js"></script>
```
Comment (ichala) [inline src/index.html:49]: - [ ] you guys have an error in your website caused by calling index.js in your index.html file, since we are using webpack, it will be responsible for bundling & calling the index.js inside the HTML file, so just remove the importing line `<script src="./index.js"></script>` and the error will be fixed 👌 

image

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,252 @@
+/* eslint-disable implicit-arrow-linebreak */
+/* eslint-disable consistent-return */
+/* eslint-disable array-callback-return */
+/* eslint-disable no-unused-expressions */
+/* eslint-disable comma-dangle */
+/* eslint-disable arrow-body-style */
```
Comment (ichala) [inline src/index.js:6]: - [ ]  linter is used to flag programming errors, bugs, stylistic errors, and suspicious constructs, so to keep your code readable and understandable by everyone, kindly remove the linters disabling commands and please fix all the linters error 🙏

**Review by ichala — CHANGES_REQUESTED** [review]
Hi  team 
Good job so far! 
👏👏👏 
There are some issues that you still need to work on to go to the next project but you're almost there!

the-simpsons-homer-simpson

## To Highlight 🎉

- Great Design✔️ 
- Good Presentation Video ✔️
- Application Works as expected ✔️

 ## Required Changes ♻️

- [ ] 🛑  **_Check The comments below_**



## Optional suggestions

- N/A

**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.html (hunk 3 above)
Comment (Baayeh) [reply src/index.html]: Thank @ichala for pointing that out. We have struggled to figure out where that was coming from🙏

**Review by dasileker — APPROVED** [review]
### **Hi @Baayeh,** 

### **Congratulations! 🎉**

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 : 
 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
